utils/value.go

```diff
@@ -21,3 +26,30 @@ func FilterMap[K comparable, V any](orig map[K]V, predicate func(K, V) bool) map
 	}
 	return ret
 }
+
+// Rand is a wrapper for either a rand.Rand or a pass-through to the shared rand.x functions.
+type Rand interface {
+	Float64() float64
+}
+
+// Testing returns true when you are running in test suite.
+func Testing() bool {
+	// TODO use official Testing method when available https://github.com/golang/go/issues/52600
```

Comment:
It's already available in 1.21.0 ([link](https://pkg.go.dev/testing@go1.21.0#Testing)). Or does this mean once we upgrade? Maybe we can add which version we're waiting to upgrade to?

Reply:
oops didn't realize it was released. fixed comment

(note our go 1.21 upgrade is two step -- base image, then go.mod, so it will be a bit before we are fully on latest)